rover_video/src/screenshot_server.hpp

```diff
@@ -0,0 +1,681 @@
+            stopRecording.store(other.stopRecording.load(std::memory_order_acquire), std::memory_order_release);
+
+            camURL = std::move(other.camURL);
+            pipeline = std::move(other.pipeline);
+            filename = std::move(other.filename);
+            videoFolderPath = std::move(other.videoFolderPath);
+            files = std::move(other.files);
+            recordingNumber = other.recordingNumber;
+            startTime = other.startTime;
+            frame_width = other.frame_width;
+            frame_height = other.frame_height;
+            fps = other.fps;
+            logger_ = std::move(other.logger_);
+
+            appender = std::move(other.appender);
+            cap = std::move(other.cap);  // Move cv ressources
+            video_writer = std::move(other.video_writer);
+            frame = std::move(other.frame);
+        }
+        return *this;
+    }
+
+    ~Recording()
+    {
+        if (cap.isOpened())  // avoid unnecessary logging when creating temporary objects
+        {
+            stopRecording.store(true);
+
+            if (recordingThread->joinable())
+            {
+                recordingThread->join();
+            }
+            // Release resources
+            this->cap.release();
+            this->video_writer.release();
+            this->appender.release();
+
+            RCLCPP_INFO(*logger_, "Recording stopped.");
+        }
+    }
+};
+
+class CameraNode : public rclcpp::Node
+{
+  public:
+  private:
+    rclcpp::Service<rover_msgs::srv::CameraControl>::SharedPtr _srv_control;
+    rclcpp::Subscription<rover_msgs::msg::GpsPosition>::SharedPtr _msg_position;
+    float last_latitude = 0.0, last_longitude = 0.0;
+
+    void controlIPCam(const rover_msgs::srv::CameraControl::Request& request, rover_msgs::srv::CameraControl::Response& response);
+    std::string getCurrentTime();
+    std::string getFileName(const std::string capture_name, std::string camURL, int state);
+    std::string getCamID(std::string cameraURL);
+    void callbackPosition(const rover_msgs::msg::GpsPosition& gps_message);
+    const std::string getFolderPath(int state);
+    bool folderExists(const std::string& path);
+    bool createFolder(const std::string& path);
+    bool getScreenshot(std::string screenshotFolderPath, std::string filename, std::string cameraURL);
+
+    bool newRecording(std::string videoFolderPath, std::string filename, std::string cameraURL);
+    bool stopRecording(std::string cameraURL);
+    bool StartWatchDog();
+    void VideoWatchDogFunction();
+    void RequestShutdown(std::string camURL);
+    std::atomic<bool> watchDogStop{false};
+    std::mutex recordingMutex;
+    std::thread videoWatchDog;
+    std::condition_variable recordingCv;
+
+    std::unordered_map<std::string, Recording> RecordingMap;
+    std::unordered_set<std::string> RecordingShutdownRequestSet;
+
+  public:
+    CameraNode();
+    ~CameraNode() {}
```

Comment:
A good way to explicit this is:


```suggestion
    CameraNode();
    ~CameraNode() = default;
```

or even better:

```suggestion
    CameraNode();
```

Reply:
There are underlying reasons (search it on the web if you really wanna know, asking ChatGPT always yield a good enough answer) but mostly, using default or not defining at all explicitely says "I wanted a default destructor", and using ~ClassName() {} feels more like "I wanted to implement some code in the destructor but I forgot".